magpie/run/magpie-run-project-hive

```diff
@@ -0,0 +1,238 @@
+#  LLNL-CODE-644248
+#  
+#  This file is part of Magpie, scripts for running Hadoop on
+#  traditional HPC systems.  For details, see https://github.com/llnl/magpie.
+#  
+#  Magpie is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# These are functions to be called by magpie-run
+
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-submission-type
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-dirs
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-user
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-defaults
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-log
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-paths
+
+# Return 0 if service up, 1 if not
+__Magpie_run_check_hive_up () {
+    cd ${HIVE_HOME}
+
+    local serverschk=0
+    local serversup=0
+    for server in `cat ${HIVE_CONF_DIR}/masters`
+    do
+        serverschk=`expr ${serverschk} + 1`
+        local serverisup=`${MAGPIE_REMOTE_CMD:-ssh} ${MAGPIE_REMOTE_CMD_OPTS} $server ps aux | grep '[H]iveServer2' | awk '{print $2}'`
+        local metastoreup=`${MAGPIE_REMOTE_CMD:-ssh} ${MAGPIE_REMOTE_CMD_OPTS} $server ps aux | grep '[H]iveMetaStore' | awk '{print $2}'`
+
+        if [ "${serverisup}X" != "X" ] \
+           && [ "${metastoreup}X" != "X" ]
+        then
+            serversup=`expr ${serversup} + 1`
+        fi
+    done
+    
+    echo "$serversup/$serverschk Hive Servers are up."
+    if [ "$serverschk" -eq "$serversup" ]
+    then
+        return 0
+    fi
+
+    return 1
+}
+
+Magpie_run_start_hive () {
+    if [ "${HIVE_SETUP}" == "yes" ] && [ "${magpie_run_prior_startup_successful}" == "true" ]
+    then
+        if [ "${magpie_run_hadoop_setup_successful}" == "0" ]
+        then
+            Magpie_output_internal_error "Attempt to setup Hive without Hadoop being setup"
+            magpie_run_hive_should_be_torndown=0
+            magpie_run_hive_setup_successful=0
+            return 1
+        fi
+
+        cd ${HIVE_HOME}
+
+        if [ ${HIVE_MODE} != "setuponly" ]
+        then
+
+            Magpie_make_all_local_dirs_node_specific
+            ${MAGPIE_SCRIPTS_HOME}/bin/magpie-postgres-daemon.sh start ${HIVE_CONF_DIR}
+
+            # Check to verify Postgres is up and running
+            $POSTGRES_HOME/bin/pg_isready -h localhost -p $PGPORT &> /dev/null
```

Comment:
localhost?  should be HIVE_MASTER_NODE?

Reply:
So this I think is something to do with either postgresql or our security here.  When using HIVE_MASTER_NODE the pg_isready fails (it prints properly, checking node:port).  Even submitting manually once everything is running if we use anything other than localhost here it fails. I'm not sure why at this point, however the only thing I can think of is that its either security related (can't have pg_isready perform a loopback?) or postgres related.